substrate/frame/support/src/transaction_extensions.rs

```diff
@@ -0,0 +1,93 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Transaction extensions.
+
+use crate::{CloneNoBound, EqNoBound, PartialEqNoBound, RuntimeDebugNoBound};
+use codec::{Codec, Decode, Encode};
+use scale_info::{StaticTypeInfo, TypeInfo};
+use sp_io::hashing::blake2_256;
+use sp_runtime::{
+	impl_tx_ext_default,
+	traits::{
+		transaction_extension::TransactionExtensionBase, DispatchInfoOf, Dispatchable,
+		IdentifyAccount, TransactionExtension, Verify,
+	},
+	transaction_validity::{InvalidTransaction, TransactionValidityError, ValidTransaction},
+};
+use sp_std::fmt::Debug;
+
+#[derive(
+	CloneNoBound, EqNoBound, PartialEqNoBound, Encode, Decode, RuntimeDebugNoBound, TypeInfo,
+)]
+#[codec(encode_bound())]
+#[codec(decode_bound())]
+pub struct VerifyMultiSignature<V: Verify>
```

Comment:
the lack of documentation does not help here with review -- no context on what this is, why it is here, where it is used. ofc I could figure all of this out myself, but it is better to provide some pointers, especially to a public type. 

From an old conov I remember this stuff was used in PoP.

Reply:
It's just the extension that would do the job of the signature checking performed in `Checkable` when we transition to general transactions.

Reply:
I updated the documentation and the extension is now used in the authorization example.